ansible_base/lib/utils/views.py

```diff
@@ -0,0 +1,84 @@
+import logging
+import time
+
+from django.conf import settings
+from django.db import connection
+from django.utils.translation import gettext_lazy as _
+from rest_framework import views
+from rest_framework.exceptions import AuthenticationFailed, ParseError, PermissionDenied, UnsupportedMediaType
+
+from ansible_base.lib.utils.settings import get_function_from_setting
+
+logger = logging.getLogger('ansible_base.lib.utils.views')
+
+
+class AnsibleBaseView(views.APIView):
+    def initialize_request(self, request, *args, **kwargs):
+        """
+        Store the Django REST Framework Request object as an attribute on the
+        normal Django request, store time the request started.
+        """
+        self.time_started = time.time()
+        if getattr(settings, 'SQL_DEBUG', False):
+            self.queries_before = len(connection.queries)
+
+        # If there are any custom headers in REMOTE_HOST_HEADERS, make sure
+        # they respect the allowed proxy list
+        if all(
+            [
+                settings.PROXY_IP_ALLOWED_LIST,
+                request.environ.get('REMOTE_ADDR') not in settings.PROXY_IP_ALLOWED_LIST,
+                request.environ.get('REMOTE_HOST') not in settings.PROXY_IP_ALLOWED_LIST,
```

Comment:
This isn't general, this is AWX-specific, and because of that we probably need some more eyes on this.

Reply:
Removed for now. We can think about pulling it back in later if needed.